gtsfm/bundle/bundle_adjustment.py

```diff
@@ -295,6 +298,16 @@ def run_ba_stage_with_filtering(
         verbose: bool = True,
     ) -> Tuple[GtsfmData, GtsfmData, List[bool], float]:
         """Runs bundle adjustment and optionally filters the resulting tracks by reprojection error."""
+        logger.info(
+            "Input: %d tracks on %d cameras", initial_data.number_tracks(), len(initial_data.get_valid_camera_indices())
+        )
+        if initial_data.number_tracks() == 0 or len(initial_data.get_valid_camera_indices()) == 0:
+            # No cameras or tracks to optimize, so bundle adjustment is not possible, return invalid result.
+            logger.error(
+                "Bundle adjustment aborting, optimization cannot be performed without any tracks or any cameras."
+            )
+            return initial_data, initial_data, [False] * initial_data.number_tracks(), 0.0
```

Comment:
could we add a comment what the last return arg represents? L373 says its for `verbose` but kinda weird its a float

Reply:
not sure what you mean, the verbose is the input arg, this float is a return value. It represents the final error after optimization, a dummy in this case. Anyway, I updated the docstring.